### ISSUE TITLE
[v9.0] Trim option is not working

### ISSUE THREAD

**Issue by Kouraman** [callstack/react-native-testing-library#882]
## Describe the bug

  After migrating from the v7.2 to the 9.0, I had some breaking change in my code
  `expect(card.queryByText('test. ')).not.toBeNull();` 
didn't passed anymore. Reading the doc, I tried to use the TextMatch options. But
`expect(card.queryByText('test. ', { exact: false })).not.toBeNull();` didn't worked either

After some research and test, I finally found that was because my string needed to be trimmed.
`expect(card.queryByText('test. '.trim())).not.toBeNull();` This one worked. 

Then, I tried to use the trim option, `expect(card.queryByText('test. ', { trim: true })).not.toBeNull();` but it didn't worked either

## Expected behavior

`expect(card.queryByText('test. ', { trim: true })).not.toBeNull();` should pass

`expect(card.queryByText('test. ', { exact: false })).not.toBeNull();` should pass to, since it's presented as a kind of compatibility mode when migrating from v7, and feel like it should be the more permissive thing possible

## Steps to Reproduce

This test should pass

```
import { render } from '@testing-library/react-native';
import { Text } from 'react-native';

describe('test', () => {
  it('should pass', () => {
    const textWithSpace = 'test. ';
    const { getByText } = render(<Text>{textWithSpace}</Text>);
    expect(getByText(textWithSpace, { trim: true })).not.toBeNull();
  });
});
```

## Versions

  npmPackages:
    @testing-library/react-native: ^9.0.0 => 9.0.0 
    react: ^17.0.2 => 17.0.2 
    react-native: ^0.66.0 => 0.66.3 
    react-test-renderer: 17.0.2 => 17.0.2

**Comment by AugustinLF** [comment]
Hmmm, interesting, the behaviour is the same as in `@testing-library/react` (you can check https://testing-playground.com/), but I'm not sure if that's correct or not. My gut feeling is that you're right, and both target and matched strings should be normalised the same way. However, I'd say we probably don't want the behaviour to differ between the two platforms. We should probably raise that with them.

**Comment by mdjastrzebski** [comment]
Closing as fixed by #984. The repo test now passes correctly.
